src/wp-includes/embed.php

```diff
@@ -1059,18 +1059,15 @@ function enqueue_embed_scripts() {
 }
 
 /**
- * Prints the CSS in the embed iframe header.
+ * Enqueue the CSS in the embed iframe header.
  *
- * @since 4.4.0
+ * @since 6.4.0
  */
-function print_embed_styles() {
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	$suffix    = SCRIPT_DEBUG ? '' : '.min';
-	?>
-	<style<?php echo $type_attr; ?>>
-		<?php echo file_get_contents( ABSPATH . WPINC . "/css/wp-embed-template$suffix.css" ); ?>
-	</style>
-	<?php
+function enqueue_embed_styles() {
+	if ( has_action( 'embed_head', 'print_embed_styles' ) ) {
+		remove_action( 'embed_head', 'print_embed_styles' );
+		wp_enqueue_style( 'wp-embed-template' );
+	}
```

Comment:
IMO, better to have the `if` condition separated off from the main code for the function.

```suggestion
	if ( ! has_action( 'embed_head', 'print_embed_styles' ) ) {
		return;
	}
	remove_action( 'embed_head', 'print_embed_styles' );
	
	wp_enqueue_style( 'wp-embed-template' );
```

Also the code conditional statement could have a comment explaining the reason for the action.